src/common/src/config.rs

```diff
@@ -1534,6 +1536,10 @@ pub mod default {
             pub fn object_store_req_retry_max_attempts() -> usize {
                 DEFAULT_RETRY_MAX_ATTEMPTS
             }
+
+            pub fn retry_unhandled_503_error() -> bool {
```

Comment:
Only handling a certain error code does not seem to be scalable. It will cause trouble when we temporarily need to add error codes to retry.

The idea is to add a configurable error code collection and retry all error codes so that we can extend it through configuration.

Reply:
AFAIK the SDK doesn't expose a 1-1 mapping to the underlying http code. They're parsed into
```
pub enum SdkError<E, R> {
    /// The request failed during construction. It was not dispatched over the network.
    ConstructionFailure(ConstructionFailure),

    /// The request failed due to a timeout. The request MAY have been sent and received.
    TimeoutError(TimeoutError),

    /// The request failed during dispatch. An HTTP response was not received. The request MAY
    /// have been sent.
    DispatchFailure(DispatchFailure),

    /// A response was received but it was not parseable according the the protocol (for example
    /// the server hung up without sending a complete response)
    ResponseError(ResponseError<R>),

    /// An error response was received from the service
    ServiceError(ServiceError<E, R>),
}
```

Reply:
And "slow down" must belong to `ServiceError `.

Reply:
get it ,  the approach I mentioned is difficult to implement.

Reply:
please add some comments for `unhandled_503`.

Reply:
Indeed `unhandled_503` is not a precise name. Renamed.